flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/metadata/FlinkRelMdSize.scala

```diff
@@ -126,7 +126,7 @@ class FlinkRelMdSize private extends MetadataHandler[BuiltInMetadata.Size] {
     // get each column's RexNode (RexLiteral, RexInputRef or null)
     val projectNodes = (0 until fieldCount).map {
       i =>
-        val initNode: RexNode = rel.getCluster.getRexBuilder.constantNull()
+        val initNode: RexNode = rel.getCluster.getRexBuilder.makeNullLiteral(rel.getRowType)
```

Comment:
Deprecated in https://github.com/apache/calcite/commit/b432756e2be9ad0557a56254550eb4438dd0efcf
in favor of `#makeNullLiteral(RelDataType)`
because 

> because it produces untyped NULL literals that make planning difficult

Reply:
Is this correct? Shouldn't it be `rel.getRowType.getField(i)`?

Reply:
good catch, yes, you're right, thanks

Reply:
done